projects/angular/src/data/datagrid/datagrid.ts

```diff
@@ -274,7 +278,19 @@ export class ClrDatagrid<T = any> implements AfterContentInit, AfterViewInit, On
           /**
            * Reopen updated row or close it
            */
-          row ? this.detailService.open(row.item, row.detailButton.nativeElement) : this.detailService.close();
+          if (row) {
+            this.detailService.open(row.item, row.detailButton.nativeElement);
+          } else if (!this.hasVirtualScroller || !row) {
+            this.detailService.close();
+          }
+        }
+
+        // retain active cell
+        const active = this.keyNavigation.getActiveCell();
+        if (active) {
+          this.zone.runOutsideAngular(() => {
+            setTimeout(() => this.keyNavigation.setActiveCell(active));
+          });
```

Comment:
When this case would happen? Because we are removing the activeCell on mouse wheel and I see it's null most of the time here.

Reply:
This is to keep the focus when navigating with `PageUp` and `PageDown` buttons. 

Comment added in commit 8f155e7.